xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp

```diff
@@ -263,16 +208,24 @@ CDVDVideoCodec::VCReturn CAddonVideoCodec::GetPicture(VideoPicture* pVideoPictur
 //    pVideoPicture->iLineSize[0] = picture.stride[0];
 //    pVideoPicture->iLineSize[1] = picture.stride[1];
 //    pVideoPicture->iLineSize[2] = picture.stride[2];
-//    pVideoPicture->iWidth = picture.width;
-//    pVideoPicture->iHeight = picture.height;
-//    pVideoPicture->pts = DVD_NOPTS_VALUE;
-//    pVideoPicture->dts = DVD_NOPTS_VALUE;
-//    pVideoPicture->color_range = 0;
-//    pVideoPicture->color_matrix = 4;
-//    pVideoPicture->iFlags = DVP_FLAG_ALLOCATED;
-//    if (m_codecFlags & DVD_CODEC_CTRL_DROP)
-//      pVideoPicture->iFlags |= DVP_FLAG_DROPPED;
-//    pVideoPicture->format = RENDER_FMT_YUV420P;
+    pVideoPicture->iWidth = picture.width;
+    pVideoPicture->iHeight = picture.height;
+    pVideoPicture->pts = picture.pts;
+    pVideoPicture->dts = DVD_NOPTS_VALUE;
+    pVideoPicture->color_range = 0;
+    pVideoPicture->color_matrix = 4;
+    pVideoPicture->iFlags = 0;
+    if (m_codecFlags & DVD_CODEC_CTRL_DROP)
+      pVideoPicture->iFlags |= DVP_FLAG_DROPPED;
+
+    m_videoBufferManager.Configure(AV_PIX_FMT_YUV420P, picture.width, picture.height, picture.stride[0], picture.height, picture.decodedDataSize);
```

Comment:
clients are not supposed to call internals. clients just request a buffer, the rest is handled internally